pandas/core/indexing.py

```diff
@@ -681,6 +681,7 @@ def __setitem__(self, key, value):
         self._has_valid_setitem_indexer(key)
 
         iloc = self if self.name == "iloc" else self.obj.iloc
+        iloc.name = self.name
```

Comment:
is this necessary?  it really shouldnt be

Reply:
oh never mind, its worse than i thought.  dont do this.

Reply:
The only option is thento pass the name through as a input, isn‘t it?

I preferred this too, but looked ugly. That was the the reason I tried this.

Reply:
I passed it through as function input now. Is this better?